numpy/distutils/fcompiler/gnu.py

```diff
@@ -122,26 +122,18 @@ def get_flags_linker_so(self):
             # error checking.
             if not target:
                 # If MACOSX_DEPLOYMENT_TARGET is not set in the environment,
-                # we try to get it first from the Python Makefile and then we
-                # fall back to setting it to 10.3 to maximize the set of
+                # we try to get it first from sysconfig and then
+                # fall back to setting it to 10.7 to maximize the set of
```

Comment:
I think this should be `10.9`, compatibility for lower versions was dropped pretty much across the board I believe (Cc @matthew-brett). Before that it was `10.6`, it's never been `10.7`.

Reply:
Updated to 10.9. I think this is never used anyway, all python builds should set the sysconfig value. Maybe we should just raise an error instead of guessing.